pandas/tests/plotting/test_groupby.py

```diff
@@ -12,6 +12,23 @@
 from pandas.tests.plotting.common import TestPlotBase
 
 
+@td.skip_if_no_mpl
+def test_hist_bins_match():
+    # GH-22222
+    N = 100
+    bins = 5
+
+    np.random.seed(0)
+    df = DataFrame(np.append(np.random.randn(N), np.random.randn(N) / 10),
+                   columns=['rand'])
+    df['group'] = [0] * N + [1] * N
+    g = df.groupby('group')['rand']
+    ax = g.hist(bins=bins, alpha=0.7, equal_bins=True)[0]
+    bin_width_group0 = ax.patches[0].get_width()
+    bin_width_group1 = ax.patches[bins].get_width()
```

Comment:
Instead of using `np.random` can you not define a set of data whereby you can also easily assert that both groups have an equal number of bins across the same X-scale?

Reply:
I can replace `np.random` with some predefined values. But I think that comparing the number of bins wouldn't do the job. That's the current state of the code. We're here just changing the range without changing the number of bins.

I've written a new test, that compares the x-axis values of the bins, that should be more rigorous. For some reason the functions see kwargs differently in python 2 and 3, so some CI tests fail in the former. I'll make a commit once I figure out what's going on.